controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -212,8 +212,7 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.
 	defer func() {
 		// Always attempt to update status.
 		if err := r.updateStatus(ctx, controlPlane); err != nil {
-			var connFailure *internal.RemoteClusterConnectionError
-			if errors.As(err, &connFailure) {
+			if errors.Is(err, &internal.RemoteClusterConnectionError{}) {
```

Comment:
I spent the time to actually test this
```
			if errors.Is(err, &internal.RemoteClusterConnectionError{}) {
				log.Error(err, "test")
			}

			var connFailure *internal.RemoteClusterConnectionError
			if errors.As(err, &connFailure) {
				log.Error(err, "Could not connect to workload cluster to fetch status")
			}
```

Turns out errors.As works, errors.Is doesn't

Do we really have to do these refactorings? If we want to go ahead I would like to see prove for every change that it still works (i.e. unit test coverag)